### DISCUSSION TITLE
課題12についての質問（7/25）

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#50] [Q&A]
①問題３のif文の行を被っているところを減らしました。
例：ABA（A勝利）とAAB（A勝利）は並び替えれば同じなので、一つを消去。

ターミナルで出力するときに、たまに「○○さんが勝ちです」と出てこなくなる時があります。Console.logを全てのifに対し書いているので、出るはずなのですが、なぜ出力されないのでしょうか？

②問題４のfor文の式を変更しました。
164行目で配列の中の人全員とmaxを比べる条件式を書きました。
そして、165行目を全員に当てはまるように${users4[ ] .name}としたのですが、[ ] の部分でエラーとなってしまいます。
ここに0や1と書いてしまうと特定の人の名前だけ出てしまうので、書いてはいけないと考えました。
このconsole.logは全員分（この問題だと４人）書かなければいけないのでしょうか？
それとも、一つのconsole.logで出力するのでしょうか？

また、全員とmaxを比べる式やconsoleでの出し方がわかりません。

**Comment by nsc-kiwamu** [comment]
①の回答
必要なパターンを消去しているのでコンソールに出ていないのだと思います。
「ABA（A勝利）とAAB（A勝利）は並び替えれば同じなので、一つを消去」
並び変えれば同じですが、並び変えていないので必要なパターンが消えてしまっているのでは？

②の回答
「164行目で配列の中の人全員とmaxを比べる条件式を書きました。」
やりたいことは合っていますが、ソースコードが合っていないですね。
今のソースコードだと
「配列の長さがmaxと同じ場合」
という条件です。

「165行目を全員に当てはまるように${users4[ ] .name}としたのですが、[ ] の部分でエラーとなってしまいます。
ここに0や1と書いてしまうと特定の人の名前だけ出てしまうので、書いてはいけないと考えました。」
おしいですね。
わざわざfor文でループしていますが、ループする度にインクリメントされて0→1→2→3となる変数が無いですか？
